src/systemd_ctypes/pathwatch.py

```diff
@@ -145,7 +145,9 @@ def close(self) -> None:
         # This is a little bit tricky: systemd doesn't have a specific close
         # API outside of unref, so let's make it as explicit as possible.
         self._watch = None
-        self._source = None
+        if self._source is not None:
+            self._source.cancel()
+            self._source = None
```

Comment:
@allisonkarlitskaya I'm not 100% sure if that actually fixes anything. I can't prove that it does, nor have a test case, I just came across it when debugging last week's 100% CPU livelock.

Reply:
I.e. I'm happy to take out this commit if you think it's nonsense/redundant.